### PR TITLE
Last preparations for 2.0.0

### PR DESCRIPTION
**Pull request by mrpixeldream**
I've added [roave/backward-compatibility-check](https://github.com/Roave/BackwardCompatibilityCheck/) to make sure we catch all bc breaks in new versions.  
In addition, I have adapted the README to make sure the recommended installation method installs the current version. 

After this PR was merged, we should be fully ready for 2.0.0.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,48 +1,54 @@
...
-    "authors": [
-        {
-            "name":     "Marco Pivetta",
-            "email":    "ocramius@gmail.com",
-            "homepage": "http://ocramius.github.io/"
-        }
-    ],
-    "require": {
-        "php": "^7.3"
+  "name": "ocramius/lazy-property",
```
Comment (Ocramius) [inline composer.json]: Hmm, consider reverting and squashing the spacing change

File: .travis.yml [hunk 2]
```diff
@@ -15,3 +15,4 @@ script:
   - ./vendor/bin/phpstan analyse --no-interaction --no-progress
   - ./vendor/bin/psalm
   - ./vendor/bin/infection
+  - ./vendor/bin/roave-backward-compatibility-check --from=1.0.1
```
Comment (Ocramius) [inline .travis.yml]: Note: this will make `master` red until we tag

Comment (mrpixeldream) [reply .travis.yml]: Ha, yes. Maybe we could add the bc check in a minor instead? Like, 2.1.0? I think that would make things cleaner. We'd have a green master and could go without `--from=1.0.1` from the beginning as well.

Comment (Ocramius) [reply .travis.yml]: We can also use ` || true` and use the output to generate release notes

Comment (mrpixeldream) [reply .travis.yml]: Does this feel clean to you? I'd rather generate them locally for now and then add the command to Travis as soon as we've tagged 2.0.0, but I'm not sure. Copying over the notes from Travis to the release or from the local command line to release doesn't really make a difference, or would you generate them completely automatically?

Comment (mrpixeldream) [reply .travis.yml]: Pushing that back to after 2.0.0 release as discussed.

**Pull request merged**